strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/services/ArtifactManagementServiceImplTest.java

```diff
@@ -359,26 +376,36 @@ public void testForceDelete(@MavenRepository(repositoryId = TFD_RELEASE_WITHOUT_
                    "when allowsForceDeletion is not enabled!");
     }
 
-    @ExtendWith(RepositoryManagementTestExecutionListener.class)
+    @ExtendWith({ RepositoryManagementTestExecutionListener.class,
+                  ArtifactManagementTestExecutionListener.class })
     @Test
     public void testRemoveTimestampedSnapshots(@MavenRepository(repositoryId = TRTS_SNAPSHOTS,
                                                                 policy = RepositoryPolicyEnum.SNAPSHOT)
-                                               Repository repository)
+                                               Repository repository,
+                                               @MavenTestArtifact(repositoryId = TRTS_SNAPSHOTS,
+                                                                  id = "org.carlspring.strongbox:timestamped-current",
+                                                                  versions = { "2.0-20190701.190020-1",
+                                                                               "2.0-20190701.190145-2",
+                                                                               "2.0-20190701.190250-3"})
+                                               List<Path> currentDateArtifactPaths,
+                                               @MavenTestArtifact(repositoryId = TRTS_SNAPSHOTS,
+                                                                  id = "org.carlspring.strongbox:timestamped-previous",
+                                                                  versions = { "2.0-20190626.190020-1",
+                                                                               "2.0-20190626.190145-2"})
+                                               List<Path> previousDateArtifactPaths)
             throws Exception
     {
-        String repositoryId = repository.getId();
+        final String storageId = repository.getStorage().getId();
+        final String repositoryId = repository.getId();
 
-        RootRepositoryPath repositoryPath = repositoryPathResolver.resolve(repository);
+        // Generate timestamp for current date.
+        String timestamp = getTimestamp(0);
 
-        // Artifact can't be created with annotation because we need the current date for the timestamp.
-        MavenArtifact artifact = createTimestampedSnapshotArtifact(repositoryPath.toString(),
-                                                                   "org.carlspring.strongbox",
-                                                                   "timestamped",
-                                                                   "2.0",
-                                                                   "jar",
-                                                                   null,
-                                                                   3);
```

Comment:
@ptirador seems it was 3 snapshot versions before, and you have you have only two versions within `@MavenTestArtifact`.